the-blue-alliance-ios/View Controllers/Match/MatchViewController.swift

```diff
@@ -39,18 +43,38 @@ class MatchViewController: MyTBAContainerViewController {
             tbaKit: tbaKit,
             userDefaults: userDefaults
         )
+        self.infoViewController.matchSummaryDelegate = self
     }
 
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
-
+    
     // MARK: - View Lifecycle
 
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
 
         Analytics.logEvent("match", parameters: ["match": match.key!])
     }
+    
+}
 
+extension MatchViewController: MatchSummaryViewDelegate {
+    
+    func teamPressed(teamNumber: Int) {
+        guard let event = match.event else { return }
+        guard let redTeamKeys = match.redAlliance?.teams?.array as? [TeamKey] else { return }
+        guard let blueTeamKeys = match.blueAlliance?.teams?.array as? [TeamKey] else { return }
+        
+        // get team key that matches the target teamNumber
+        guard let teamKey = (redTeamKeys + blueTeamKeys).filter({
+            $0.teamNumber == "\(teamNumber)"
+        }).first else { return }
+        
+        let teamAtEventVC = TeamAtEventViewController(teamKey: teamKey, event: event, myTBA: self.myTBA, showDetailEvent: true, showDetailTeam: true, statusService: self.statusService, urlOpener: self.urlOpener, persistentContainer: self.persistentContainer, tbaKit: self.tbaKit, userDefaults: self.userDefaults)
+        self.navigationController?.pushViewController(teamAtEventVC, animated: true)
```

Comment:
I think we can drop the `self` from here - I know I have a lot of places where I call `self.navigationController`, I write a lot of Obj-C still 😛